libpod/oci_linux.go

```diff
@@ -461,8 +462,21 @@ func (r *OCIRuntime) createOCIContainer(ctr *Container, cgroupParent string, res
 			return errors.Wrapf(define.ErrInternal, "container create failed")
 		}
 		ctr.state.PID = ss.si.Pid
-		if cmd.Process != nil {
-			ctr.state.ConmonPID = cmd.Process.Pid
```

Comment:
do we still need `cmd.Process.Pid` when `ctr.config.ConmonPidFile == ""`?

Reply:
We didn't get a correct PID in this case (it was always off because of the double-fork). We won't be able to get a PID for old containers without ConmonPidFile set, but no PID is better than reporting an incorrect PID.